api/src/main/java/org/infinispan/api/Infinispan.java

```diff
@@ -0,0 +1,44 @@
+package org.infinispan.api;
+
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.api.reactive.KeyValueStore;
+import org.infinispan.api.reactive.KeyValueStoreConfig;
+
+/**
+ * Each infinispan is a node instance. It can be embedded or client node, depending on the access point. {@link
+ * InfinispanClient} or {@link InfinispanEmbedded}
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+@Experimental
+public interface Infinispan {
+
+   /**
+    * Gets the {@link KeyValueStore} by name.
+    * <p>
+    * If the store does not exist, creates a {@link KeyValueStore} with the default values
+    *
+    * @param name, name of the store
+    * @return {@link KeyValueStore}
+    */
+   <K, V> KeyValueStore<K, V> getKeyValueStore(String name);
```

Comment:
What are the guarantees of this call with embedded and client? For example, in client mode, will this call only return once a successful connection has been made to the server? We have to be careful not to make this a blocking call that depends on IO result. The same happens with embedded and start/join of JGroups with a cluster.

Reply:
I'm doing async